src/Browser/Avalonia.Browser/BrowserInputHandler.cs

```diff
@@ -190,8 +190,9 @@ public bool OnDragEvent(string type, double offsetX, double offsetY, int modifie
 
         var position = new Point(offsetX, offsetY);
 
-        effectAllowedStr ??= "none";
+        var effectAllowedStr = dataTransfer.GetPropertyAsString("effectAllowed") ?? "none";
```

Comment:
Sync interop from C# to JS might break WasmEnableThreads on browser.
The tricky part is that .NET 9+ runtime runs C# code in a dedicated worker thread. And any JS interoperability, especially one that access browser APIs (like here), will invoke interop between threads. It was safer to prepare as much as possible parameters on the JS side and pass them to .NET

Reply:
Though, to be fair, I am not sure if old dnd was working with multithreading. It might have been broken even before.